scripts/zones/Castle_Zvahl_Baileys/mobs/Dark_Spark.lua

```diff
@@ -4,9 +4,16 @@
 -- Involved in Quests: Borghertz's Hands (AF Hands, Many job)
 -- !pos 63 -24 21 161
 -----------------------------------
-require("scripts/globals/settings");
+require("scripts/globals/keyitems")
 -----------------------------------
 
 function onMobDeath(mob, player, isKiller)
-    player:setCharVar("BorghertzSparkKilled",1);
-end;
\ No newline at end of file
+    if
+        player:getCharVar("BorghertzSparkKilled") == 0 and
```

Comment:
mind if we change this to a local var same as the torch ID, if you logout with out touchng you would lose the torch refrence anyway. this particular var check on this line really isnt needed though since we have the other 2 checks.

Reply:
if you get DC'd do you have to repeat the fight though?

Reply:
Would have to anyway becuase it's also checking a local var vet on the torch on popping it and retrieving

Reply:
I mean: whats retail do.

just..we're here anyway. ¯\\\_(ツ)\_/¯

Reply:
Agree that we should mimic retail.  If someone with access can tell me how it works there, I can code to match.

Until then, how about this new commit?  This saves the torch used to spawn the crediting Spark to each player on the quest who helps kill it.  This way, the particular torch will persist per-player across logouts, or if someone else comes and kills a different Dark Spark before original player has a chance to touch the torch.

Reply:
Agree that we should mimic retail.  If someone with access can tell me how it works there, I can code to match.

Until then, how about this new commit?  This saves the torch used to spawn the crediting Spark to each player on the quest who helps kill it.  This way, the particular torch will persist per-player across logouts, or if someone else comes and kills a different Dark Spark before original player has a chance to touch the torch.